realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -831,15 +824,17 @@ public void removeChangeListeners() {
      * Notifies all registered listeners.
      */
     void notifyChangeListeners() {
-        // table might be null (if the async query didn't complete
-        // but we have already registered listeners for it)
-        if (pendingQuery != null && !isCompleted) return;
-
-        long version = table.sync();
-        if (currentTableViewVersion != version) {
-            currentTableViewVersion = version;
-            for (RealmChangeListener listener : listeners) {
-                listener.onChange();
+        if (listeners != null && !listeners.isEmpty()) {
+            // table might be null (if the async query didn't complete
+            // but we have already registered listeners for it)
+            if (pendingQuery != null && !isCompleted) return;
+
+            long version = table.sync();
```

Comment:
That's a good point, but all calls to `RealmResults#createFromTableOrView`, `RealmResults#createFromDynamicTableOrView` are using `TableView`.  

`RealmResults#createFromTableQuery` & `RealmResults#createFromDynamicClass`  are used by async queries that will call `importHandoverTableView` to populate `Table` with a `TableView`